src/EditorFeatures/CSharpTest/Classification/SyntacticTaggerTests.cs

```diff
@@ -35,21 +35,35 @@ public class SyntacticTaggerTests
                     null,
                     new SyntacticClassificationTaggerProvider(null, null, null));
 
-                SnapshotSpan span = default(SnapshotSpan);
+                var expectedLength = subjectBuffer.CurrentSnapshot.Length;
+                var actualVersionNumber = int.MaxValue;
+                var actualLength = int.MaxValue;
+                var didRelease = false;
                 tagComputer.TagsChanged += (s, e) =>
                 {
-                    span = e.Span;
-                    checkpoint.Release();
+                    actualVersionNumber = e.Span.Snapshot.Version.VersionNumber;
+                    actualLength = e.Span.Length;
+                    didRelease = checkpoint.TryRelease();
```

Comment:
Yeah, that's the tricky bit - I'm not sure the stack is even enough, since these are dispatched operations.  Any ideas on decent ways to track the actual causality?